test/presentation/shared_widgets/participant_avatars_test.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:collaction_app/presentation/shared_widgets/participant_avatars.dart';
+import 'package:collaction_app/application/participation/top_participants/top_participants_bloc.dart';
+import 'package:collaction_app/presentation/shared_widgets/shimmers/top_participants_shimmer.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:get_it/get_it.dart';
+import 'package:mocktail/mocktail.dart';
+
+import '../../test_helper.dart';
+import '../../application/participation/top_participants/top_participants_bloc.mocks.dart';
+import '../../test_utilities.dart';
+
+void main() {
+  late TopParticipantsBloc topParticipantsBloc;
+
+  setUpAll(() {
+    // Participants Bloc
+    topParticipantsBloc = MockTopParticipantsBloc();
+    when(() => topParticipantsBloc.state)
+        .thenAnswer((_) => TopParticipantsState.fetched(tTopParticipants));
+
+    GetIt.I.registerSingleton<TopParticipantsBloc>(topParticipantsBloc);
+
+    // dot env
+    dotenv.testLoad(fileInput: tDotEnv);
+  });
+
+  tearDownAll(() {
+    GetIt.I.unregister<TopParticipantsBloc>();
+  });
+  group('ParticipantAvatars tests:', () {
+    testWidgets('can render initial state', (WidgetTester tester) async {
+      when(() => topParticipantsBloc.state)
+          .thenAnswer((_) => TopParticipantsState.initial());
+
+      await buildAndPump(
+          tester: tester,
+          widget:
+              TopParticipantAvatars(crowdActionId: tCrowdactionNoPassword.id));
+      await tester.pumpAndSettle();
+
+      expect(find.byType(TopParticipantAvatars), findsOneWidget);
+    });
+
+    testWidgets('can render fetching state', (WidgetTester tester) async {
+      when(() => topParticipantsBloc.state)
+          .thenAnswer((_) => TopParticipantsState.fetching());
+
+      await buildAndPump(
+          tester: tester,
+          widget:
+              TopParticipantAvatars(crowdActionId: tCrowdactionNoPassword.id));
+      await tester.pump();
+
+      expect(find.byType(TopParticipantsShimmer), findsOneWidget);
+    });
+
+    testWidgets('fetched 1 participants', (WidgetTester tester) async {
+      when(() => topParticipantsBloc.state).thenAnswer(
+          (_) => TopParticipantsState.fetched(List.filled(1, tParticipation)));
```

Comment:
List.filled seems to be a bit overkill when you can just do `[tParticipation]`

Reply:
Changed, just had it there to match the subsequent tests